### ISSUE TITLE
genus/differentia aka Aristotelian definitions

### ISSUE THREAD

**Issue by turbomam** [berkeleybop/artificial-intelligence-ontology#75]
will require human review

**Comment by turbomam** [comment]
I submitted `src/ontology/aio-src.csv` and the following prompt to GPT 4

> This is a ROBOT template for generating an OWL ontology. Change the values in the Description column to follow the Aristotelian genus/differentia style. The label of the parent should be used as the genus.

then

> save the changes back to a CSV file?

**Comment by turbomam** [comment]
that's terrible

**Comment by turbomam** [comment]
> let's start over. do not try to implement a solution with code. just use your large language model capabilities to combine the existing description and the label of the asserted parent class into a new description that flows well linguistically

**Comment by turbomam** [comment]
> remove all of the "...known as..." fragments

**Comment by turbomam** [comment]
[aio-updated-final.csv](https://github.com/user-attachments/files/15771047/aio-updated-final.csv)

GPT is pretty confident of itself with that file name

**Comment by turbomam** [comment]
would still need a lot of manual editing. will try Claude again now.

**Comment by turbomam** [comment]
The Claude web interface doesn't want to provide a downloadable CSV and the output is too long to emit all at once. So I have asked for output containing only the ID and the new description.

**Comment by turbomam** [comment]
I think I may still have to do it in batches or use `llm`

**Comment by turbomam** [comment]
last full row:

> AIO:Convolution2DTransposeLayer,A layer that implements transposed 2D convolution sometimes called deconvolution.

**Comment by turbomam** [comment]
Some duplicates:

>AIO:LazyBatchNorm1DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.
AIO:LazyBatchNorm2DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.
AIO:LazyBatchNorm3DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.

**Comment by turbomam** [comment]
So I prompted:

>
I see some duplicates in there, like "AIO:LazyBatchNorm1DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.AIO:LazyBatchNorm2DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.AIO:LazyBatchNorm3DLayer,A batch normalization layer with lazy initialization of the num_features argument that is inferred from the input size.". Please do your best to ensure unique descriptions.

**Comment by turbomam** [comment]
Claude's generation of Aristotelian descriptions completed and I added it as a new column in the google sheet

**Comment by turbomam** [comment]
Now I loaded the sheet into GPT 4 and asked

>determine the lexical differences between the "Legacy Description" and "Claude opus Aristotelian definition" columns. I would use a jacquard or k-mer approach, but I trust you to choose what's best. generate a tsv file with tow columns: the ID and the difference.

**Comment by turbomam** [comment]
I had to convince it that I wanted a numerical diffrence score, but it finally did emit them and they look reasonable, as if a formula was applied. 

GPT 4 ultimately said that it was going to use (1 - (jaccard similarity))

I added the column and set the sheet up with a filter view and made some of the columns narrower and wrapped.

It looks like some of the description were really long and have content that should be extracted out into more comments.
